bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -0,0 +1,285 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.automower.internal.things;
+
+import static org.openhab.binding.automower.internal.AutomowerBindingConstants.*;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.automower.internal.AutomowerBindingConstants;
+import org.openhab.binding.automower.internal.actions.AutomowerActions;
+import org.openhab.binding.automower.internal.bridge.AutomowerBridge;
+import org.openhab.binding.automower.internal.bridge.AutomowerBridgeHandler;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.Mower;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AutomowerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AutomowerHandler extends BaseThingHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_AUTOMOWER);
+    private static final String NO_ID = "NO_ID";
+    private static final long DEFAULT_COMMAND_DURATION_MIN = 60;
+    private static final long DEFAULT_POLLING_INTERVAL_S = TimeUnit.MINUTES.toSeconds(10);
+
+    private final Logger logger = LoggerFactory.getLogger(AutomowerHandler.class);
+    private AtomicReference<String> automowerId = new AtomicReference<String>(NO_ID);
```

Comment:
This is only updated during initialization, why do you need it to be atomic?

Reply:
Probably because I am not familiar how the framework works from a threading perspective. As I was not sure if the initialize and dispose methods are called by the same thread I assumed they might not.

To make sure a different thread would see the most recent value of the variable I would need some kind of synchronization. Volatile would be sufficient in that case but I thought using an AtomicReference would be nicer and would not hurt.